pkg/client/fake/doc.go

```diff
@@ -17,17 +17,24 @@ limitations under the License.
 /*
 Package fake provides a fake client for testing.
 
-Deprecated: please use pkg/envtest for testing. This package will be dropped
-before the v1.0.0 release.
-
-An fake client is backed by its simple object store indexed by GroupVersionResource.
+fake client is backed by its simple object store indexed by GroupVersionResource.
 You can create a fake client with optional objects.
 
 	client := NewFakeClient(initObjs...) // initObjs is a slice of runtime.Object
 
 You can invoke the methods defined in the Client interface.
 
-When it doubt, it's almost always better not to use this package and instead use
+When in doubt, it's almost always better not to use this package and instead use
 envtest.Environment with a real client and API server.
+
+Current Limitations / Known Issues with the fake Client:
+- this client does not use reactors so it can not be setup with detailed responses for testing
```

Comment:
```suggestion
```

We can probably remove this line, given that it's clear it's a different implementation.

Reply:
The use case here is if you would like to test error cases in your controller's reconcile logic, we need a way to inject a specific error. For e.g. to test the psuedo code below:

Reply:
Sounds good to me